### Kaabachi Alaa
### 3BD1

#### Étape 1 : Charger et examiner les données

In [2]:
# Charger les bibliothèques nécessaires
library(readxl)
library(nnet)
library(caret)

#Charger Les données de Exercice1
data <- read_excel("C:\\Users\\alaak\\Downloads\\exercice 2.xlsx", sheet="Feuil1")
#Afficher les premieéres lignes pour examiner les données
head(data)
#Vérifier les valuers manquantes
colSums(is.na(data))
#Supprimer les lignes avec des valeurs manquantes
data <- na.omit(data)


Warning message:
"package 'readxl' was built under R version 4.4.2"
Warning message:
"package 'nnet' was built under R version 4.4.2"
Warning message:
"package 'caret' was built under R version 4.4.2"
Loading required package: ggplot2

Warning message:
"package 'ggplot2' was built under R version 4.4.2"
Loading required package: lattice



Continente Dis Mag Value nombre total de décès Total Affecté
1 2          8               12                      40       
2 3          7              476                  141277       
3 2          7                3                      15       
4 4          7             1250                  134250       
5 3          7               31                     200       
6 1          8              100                    2000       
  Total des dommages ('000 US $)
1 500000                        
2 100000                        
3   7310                        
4   6000                        
5   3150                        
6  25000

Continente                  Dis Mag Value 
                             0                              0 
         nombre total de décès                  Total Affecté 
                             0                              0 
Total des dommages ('000 US $) 
                             0

### 2. Expliquer la variable continente  en fonction des variables


In [4]:
model <- multinom(Continente ~ .,data=data)
summary(model)

# weights:  30 (20 variable)
initial  value 173.819295 
iter  10 value 165.148541
iter  20 value 131.900368
iter  30 value 125.614527
iter  40 value 125.098755
iter  50 value 124.569999
iter  60 value 123.846741
iter  70 value 122.842655
iter  80 value 122.716010
final  value 122.714668 
converged


Call:
multinom(formula = Continente ~ ., data = data)

Coefficients:
  (Intercept) `Dis Mag Value` `nombre total de décès` `Total Affecté`
2   -4.807929       0.6177011           -3.017864e-05    9.305018e-07
3    5.326481      -0.9018543           -3.933559e-04    1.298160e-06
4    7.680986      -1.4427079           -1.203217e-05    1.060519e-06
5   -6.456925       0.8655634           -2.439808e-02   -1.408477e-04
  `Total des dommages ('000 US $)`
2                     4.043856e-07
3                     2.230761e-07
4                     2.370130e-07
5                    -1.795839e-04

Std. Errors:
   (Intercept) `Dis Mag Value` `nombre total de décès` `Total Affecté`
2 3.457665e-08    2.398805e-07            3.356182e-05    7.155007e-07
3 9.207998e-08    6.071898e-07            3.224619e-04    1.013857e-06
4 1.374329e-08    8.091810e-08            4.799140e-05    1.007641e-06
5 3.854395e-08    2.675433e-07            4.320014e-07    2.195122e-04
  `Total des dommages ('000 US $)`
2 

### 3.Determiner la Qualité du modéle

In [5]:
#D'abord on va faire une prédiction
predictions<-predict(model,newdata=data)
#creer la matrice de confusion
MatriceDeConfusion<-table(data$Continente,predictions)
print(MatriceDeConfusion)
#Calculer la qualité du modéle:somme de diagonale / somme de matrice
print(paste("Qualité du modèle :", sum(diag(MatriceDeConfusion)) / sum(MatriceDeConfusion)))

   predictions
     1  2  3  4  5
  1 29  7  5  0  0
  2 18 13  1  0  0
  3 13  3  6  0  0
  4  6  1  2  2  0
  5  2  0  0  0  0
[1] "Qualité du modèle : 0.462962962962963"


### 4. Expliquer la variable  Dis Mag en fonction des variables


In [14]:
model <- multinom(`Dis Mag Value` ~ `nombre total de décès` +`Total Affecté`+`Total des dommages ('000 US $)`,data=data)
summary(model)

# weights:  30 (20 variable)
initial  value 193.510023 
iter  10 value 178.341875
iter  20 value 133.918263
iter  30 value 127.265709
iter  40 value 126.230292
iter  50 value 125.705007
iter  60 value 125.098559
iter  70 value 124.351402
iter  80 value 124.027365
iter  90 value 123.578972
iter 100 value 123.309541
final  value 123.309541 
stopped after 100 iterations


Call:
multinom(formula = `Dis Mag Value` ~ `nombre total de décès` + 
    `Total Affecté` + `Total des dommages ('000 US $)`, data = data)

Coefficients:
  (Intercept) `nombre total de décès` `Total Affecté`
4   1.4144650           -0.5933362445    6.511416e-06
5   0.2818687           -0.0007285482    1.874325e-05
6   1.3400011            0.0001790053    4.001649e-05
7   1.7559870            0.0001754416    4.003299e-05
8   0.4755223            0.0002000724    4.225506e-05
  `Total des dommages ('000 US $)`
4                     0.0003920540
5                     0.0003997089
6                     0.0003993648
7                     0.0003994797
8                     0.0003992066

Std. Errors:
   (Intercept) `nombre total de décès` `Total Affecté`
4 6.931299e-09            9.498022e-09    1.351225e-04
5 2.678236e-09            3.222084e-07    9.039424e-05
6 4.078630e-09            1.940440e-05    8.802337e-05
7 6.059144e-09            1.818987e-05    8.802112e-05
8 1.872733e-09         

### 5.Determiner la Qualité du modéle

In [15]:
#D'abord on va faire une prédiction
predictions<-predict(model,newdata=data)
#creer la matrice de confusion
MatriceDeConfusion<-table(data$ `Dis Mag Value`,predictions)
print(MatriceDeConfusion)
#Calculer la qualité du modéle:somme de diagonale / somme de matrice
print(paste("Qualité du modèle :", sum(diag(MatriceDeConfusion)) / sum(MatriceDeConfusion)))

   predictions
     1  4  5  6  7  8
  1  0  0  0  0  1  0
  4  0  0  0  0  1  0
  5  0  0  1  0  7  0
  6  0  0  0  0 29  1
  7  0  0  0  0 48  0
  8  0  0  0  0 14  6
[1] "Qualité du modèle : 0.509259259259259"


#### 6.Etudier la stationarité du Dis Mag value

In [26]:
# Charger la bibliothèque tseries
library(tseries)
# effectuter le test de stationnarité
DisMagValueStat = adf.test(data$ `Dis Mag Value`,alternative = "stationary")
print(DisMagValueStat)

Warning message in adf.test(data$`Dis Mag Value`, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$`Dis Mag Value`
Dickey-Fuller = -4.2359, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



p-value = 0.01 < 0.05, on peut conclure que Dis Mag Value est stationnaire d'ordre 0.



#### 7.Etudier la stationarité du nombre total de décès

In [27]:
# effectuter le test de stationnarité
NombreTotalDeDécésStat = adf.test(data$ `nombre total de décès`,alternative = "stationary")
print(NombreTotalDeDécésStat)

Warning message in adf.test(data$`nombre total de décès`, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$`nombre total de décès`
Dickey-Fuller = -4.8164, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



p-value = 0.01 < 0.05, on peut conclure que nombre total de décès est stationnaire d'ordre 0.

#### 8.Etudier la stationarité du  total affecté

In [29]:
# effectuter le test de stationnarité
TotalAffecté = adf.test(data$ `Total Affecté`,alternative = "stationary")
print(TotalAffecté)

Warning message in adf.test(data$`Total Affecté`, alternative = "stationary"):
"p-value smaller than printed p-value"



	Augmented Dickey-Fuller Test

data:  data$`Total Affecté`
Dickey-Fuller = -4.0487, Lag order = 4, p-value = 0.01
alternative hypothesis: stationary



p-value = 0.01 < 0.05, on peut conclure que le nombre total de décès est stationnaire d'ordre 0.

#### 9.Etudier la stationarité du  total affecté

In [ ]:
# effectuter le test de stationnarité
TotalDesDommagesStat = adf.test(data$ `Total des dommages ('000 US $)`,alternative = "stationary")
print(TotalDesDommagesStat)

p-value = 0.01 < 0.05, on peut conclure que le nombre total des dommages est stationnaire d'ordre 0.

### 10.Expliquer la variable Dis Mag value en fonction des variables nombre total de décès, total affecté et total des dommages.

On a tous les valeurs sont d'order 0 (p value<0.05) Alors on va Utiliser le modéle VAR.

In [70]:
library('vars')
data1 = data[, c('Dis Mag Value', 'nombre total de décès', 'Total Affecté', "Total des dommages ('000 US $)")]
model = VAR(data1,p = 1)
equation_dis_mag_value <- coef(model)[1]
print(equation_dis_mag_value)


$Dis.Mag.Value
                                       Estimate   Std. Error     t value
Dis.Mag.Value.l1                   5.184722e-03 1.027086e-01  0.05047993
nombre.total.de.décès.l1           5.259005e-06 4.425041e-06  1.18846454
Total.Affecté.l1                  -3.194770e-09 1.670082e-07 -0.01912942
Total.des.dommages...000.US....l1 -1.834996e-08 4.965101e-08 -0.36957889
const                              6.618974e+00 6.870907e-01  9.63333429
                                      Pr(>|t|)
Dis.Mag.Value.l1                  9.598387e-01
nombre.total.de.décès.l1          2.374102e-01
Total.Affecté.l1                  9.847752e-01
Total.des.dommages...000.US....l1 7.124625e-01
const                             5.279120e-16



## Interprétation :
### Dis.Mag.Value.l1 (lag 1 de Dis.Mag.Value) :
   - Estimate: 0.0052
   - Std. Error: 0.1027
   - t value: 0.0505
   - Pr(>|t|): 0.9598
   
   L'estimation du coefficient est de 0.0052, ce qui indique une relation positive mais très faible entre la valeur passée de `Dis.Mag.Value` (lag 1) et la valeur actuelle. Cependant, le p-value est de 0.96, ce qui est bien au-dessus du seuil de 0.05, indiquant que cette relation n'est pas statistiquement significative.

### nombre.total.de.décès.l1 (lag 1 de nombre total de décès) :
   - Estimate: 5.259e-06
   - Std. Error: 4.425e-06
   - t value: 1.1885
   - Pr(>|t|): 0.2374
   
   Le coefficient de 5.259e-06 suggère une relation positive faible avec la variable `Dis.Mag.Value`, mais le p-value de 0.237 est également bien supérieur à 0.05, ce qui indique que cette relation n'est pas statistiquement significative.

### Total.Affecté.l1 (lag 1 de Total Affecté) :
   - Estimate: -3.195e-09
   - Std. Error: 1.670e-07
   - t value: -0.0191
   - Pr(>|t|): 0.9848
   
   Le coefficient très faible et négatif de -3.195e-09 indique une relation quasiment nulle entre `Total.Affecté` passé et la valeur actuelle de `Dis.Mag.Value`. Le p-value de 0.985 montre que cette relation est non significative.

### Total.des.dommages...000.US....l1 (lag 1 de Total des dommages) :
   - Estimate: -1.835e-08
   - Std. Error: 4.965e-08
   - t value: -0.3696
   - Pr(>|t|): 0.7125
   
   Le coefficient est très faible et négatif, ce qui suggère une relation très faible entre les dommages passés et `Dis.Mag.Value`. Le p-value de 0.712 est bien supérieur à 0.05, ce qui indique que la relation n'est pas statistiquement significative.

### const (constante) :
   - Estimate: 6.618974
   - Std. Error: 0.6871
   - t value: 9.633
   - Pr(>|t|): 5.279e-16

   L'estimation de la constante est de 6.619, et elle est statistiquement significative avec un p-value extrêmement faible (5.279e-16). Cela suggère que, même si les autres variables explicatives ne sont pas significatives, la valeur constante a une contribution significative dans la prédiction de `Dis.Mag.Value`.

